kong/tools/queue.lua

```diff
@@ -506,6 +514,21 @@ local function enqueue(self, entry)
     return nil, "entry must be a non-nil Lua value"
   end
 
+
+  if self.concurrency == 0 then
```

Comment:
I would add a small comment here that explains that the code below is just executing the handler right away instead of actually "enqueueing" which is the desired behavior when there is no limit set for the concurrency.

Reply:
Done.